### PR TITLE
converter-driven changes

### PR DESCRIPTION
**Pull request by xeno-by**

### CODE REVIEW

**Comment by olafurpg** [comment]
Is it worth renaming the import fields now? The roles I won't miss 😺

**Comment by olafurpg** [comment]
The CI has now been fixed so the build should pass.

**Comment by xeno-by** [comment]
@olafurpg Since field names have become a public API, I think it is important to polish them.

File: changelog/1.3.0.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+## Trees
+
+  * Type.Compound has been split into `Type.With` and `Type.Refine` in order to provide common ground for compatibility with Dotty. Dotty gets rid of `with` in favor of intersection types, but it still has type refinements, so we needed to extract `Type.Refine`, so that it can be used in both dialects.
+
+    ```
+    scala> t"A with B".structure
+    res0: String = Type.With(Type.Name("A"), Type.Name("B"))
+
+    scala> t"{ def x: Int }".structure
```
Comment (olafurpg) [inline changelog/1.3.0.md]: Why is it `Type.Refine` and not `Type.Structural`?

Comment (jvican) [inline changelog/1.3.0.md]: I think it's bencause the spec considers them *Refinements*: http://scala-lang.org/files/archive/spec/2.11/03-types.html#compound-types

Comment (xeno-by) [inline changelog/1.3.0.md]: The thing in curly braces is called type refinement in the Scala Language Specification.

**Pull request merged**